homeassistant/components/switchbee/__init__.py

```diff
@@ -0,0 +1,160 @@
+"""The SwitchBee Smart Home integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from switchbee.api import CentralUnitAPI, SwitchBeeError
+from switchbee.device import DeviceType
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    CONF_DEFUALT_ALLOWED,
+    CONF_DEVICES,
+    CONF_SWITCHES_AS_LIGHTS,
+    DOMAIN,
+    SCAN_INTERVAL_SEC,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up SwitchBee Smart Home from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    central_unit = entry.data[CONF_HOST]
+    user = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    devices_map: dict[str, DeviceType] = {s.display: s for s in DeviceType}
+    allowed_devices = [
+        devices_map[device]
+        for device in entry.options.get(CONF_DEVICES, CONF_DEFUALT_ALLOWED)
+    ]
+    scan_interval = entry.options.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL_SEC)
```

Comment:
See comment in config flow, this should be fixed, set it from the const and not from config flow
```suggestion
```